atlasdb-persistent-lock-api/src/main/java/com/palantir/atlasdb/persistentlock/PersistentLockService.java

```diff
@@ -34,24 +32,22 @@
      * Attempt to acquire the lock.
      * Call this method before performing any destructive operations.
      * @param reason the reason for the lock, for logging purposes (e.g. "sweep")
-     * @return a {@link LockEntry} on success. The LockEntry will contain the given reason, and a unique ID. It is
-     *   essential that you retain a reference to this lock, as you will need it in order to release the lock.
+     * @return a {@link PersistentLockId} on success that represents the unique id of the lock that has
+     *   been acquired. You will need to provide this lock id when releasing the lock.
      */
     @POST // This has to be POST because we can't allow caching.
     @Path("acquire-backup-lock")
     @Produces(MediaType.APPLICATION_JSON)
-    LockEntry acquireBackupLock(@QueryParam("reason") String reason);
+    PersistentLockId acquireBackupLock(@QueryParam("reason") String reason);
 
     /**
      * Release a lock that you have previously acquired.
      * Call this method as soon as you no longer need the lock (e.g. because you finished deleting stuff).
-     * @param lockEntry the {@link LockEntry} you were given when you called {@link #acquireBackupLock(String)}
-     * @return A success message if the lock was successfully released
-     * @throws CheckAndSetException if there was a conflict.
+     * @param lockId the {@link PersistentLockId} you were given when you called {@link #acquireBackupLock(String)}
      */
     @POST
     @Path("release")
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
-    String releaseLock(LockEntry lockEntry) throws CheckAndSetException;
+    void releaseLock(PersistentLockId lockId);
```

Comment:
I think path params are nice for restful APIs. Large changes would be required to make this restful though